src/entries/popup/pages/messages/SignMessage/index.tsx

```diff
@@ -64,14 +64,23 @@ export function SignMessage({
     const requestPayload = getSigningRequestDisplayDetails(request);
     if (!requestPayload.msgData || !requestPayload.address || !selectedWallet)
       return;
-    const { type } = await wallet.getWallet(selectedWallet);
+    const { type, vendor } = await wallet.getWallet(selectedWallet);
     let result = null;
 
     setLoading(true);
     try {
       // Change the label while we wait for confirmation
       if (type === 'HardwareWalletKeychain') {
         setWaitingForDevice(true);
+        analytics.track(event.dappTransactionSubmitted, {
```

Comment:
Technically not submitted here; just queued for the device. Is there a reason we don't want to use the existing `dappPromptSendTransaction`, `dappPromptSignMessage`, and `dappPromptSignTypedData` events? Could just append the hardware wallet type to them

Reply:
I was just trying to make less events. Happy to add them there tho!